ДЗ-14 Введение в базы данных

berestov5 opened the pull request.

berestov5 committed:
ДЗ-14 Введение в базы данных

dmitry-bizin commented on src/main/java/ru/hogwarts/school/service/FacultyService.java:
```diff
@@ -2,45 +2,39 @@
-        }
-        faculties.put(faculty.getId(), faculty);
-        return faculty;
+        return facultyRepository.save(faculty);
```
Чем этот метод отличается от метода сохранения? Ничем. А должен отличаться, потому что это разные операции. Если нет записи с таким id, то в конечном счёте пользователю нужно вернуть 404, а иначе обновить запись

dmitry-bizin commented on src/main/java/ru/hogwarts/school/service/StudentService.java:
```diff
@@ -2,45 +2,34 @@
-        }
-        students.put(student.getId(), student);
-        return student;
+        return studentRepository.save(student);
```
Чем этот метод отличается от метода сохранения? Ничем. А должен отличаться, потому что это разные операции. Если нет записи с таким id, то в конечном счёте пользователю нужно вернуть 404, а иначе обновить запись

berestov5 committed:
ДЗ-14 Введение в базы данных
Исправил метод editFaculty и editStudent в сервисах

dmitry-bizin commented on src/main/java/ru/hogwarts/school/service/FacultyService.java:
```diff
@@ -26,15 +25,18 @@ public Faculty findFaculty(long id) {
+        if (facultyRepository.existsById(faculty.getId())) {
+            return facultyRepository.save(faculty);
+        }
+        return null;
```
выше в контроллере null не обрабатывается в 404, а просто возвращается клиенту

dmitry-bizin commented on src/main/java/ru/hogwarts/school/service/StudentService.java:
```diff
@@ -22,9 +22,13 @@ public Student findStudent(long id) {
+        if (studentRepository.existsById(student.getId())) {
+            return studentRepository.save(student);
+        }
+        return null;
```
выше в контроллере null не обрабатывается в 404, а просто возвращается клиенту

berestov5 committed:
ДЗ-14 Введение в базы данных
Исправил метод editFaculty и editStudent в сервисах 404 ошибка в сервисах добавил при null
